src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+            for (Feature feature : anomalyDetector.getFeatureAttributes()) {
+                ParseUtils.parseAggregators(feature.getAggregation().toString(), xContent, feature.getId());
+                XContentParser parser = XContentType.JSON
+                    .xContent()
+                    .createParser(xContent, LoggingDeprecationHandler.INSTANCE, feature.getAggregation().toString());
+                parser.nextToken();
+                List<String> fieldNames = parseAggregationRequest(parser);
+                BoolQueryBuilder boolQuery2 = QueryBuilders
+                    .boolQuery()
+                    .filter(rangeQuery)
+                    .filter(anomalyDetector.getFilterQuery())
+                    .filter(QueryBuilders.existsQuery(fieldNames.get(0)));
+                SearchSourceBuilder internalSearchSourceBuilder = new SearchSourceBuilder().query(boolQuery2).size(1).terminateAfter(1);
+                SearchRequest searchRequest = new SearchRequest(anomalyDetector.getIndices().get(0)).source(internalSearchSourceBuilder);
+                client.search(searchRequest, ActionListener.wrap(searchResponse -> {
+                    featureCounter.incrementAndGet();
+                    onFeatureAggregationValidation(searchResponse, feature, featureCounter);
+                }, exception -> onFailure(exception)));
+            }
+        }
+    }
+
+    private List<String> parseAggregationRequest(XContentParser parser) throws IOException {
+        List<String> fieldNames = new ArrayList<>();
+        XContentParser.Token token;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                final String field = parser.currentName();
+                switch (field) {
+                    case "field":
+                        parser.nextToken();
+                        fieldNames.add(parser.textOrNull());
+                        break;
+                    default:
+                        parser.skipChildren();
+                        break;
+                }
+            }
+        }
+        return fieldNames;
+    }
+
+    private void onFeatureAggregationValidation(SearchResponse response, Feature feature, AtomicInteger counter) throws IOException {
+        if (response.getHits().getTotalHits().value <= 0) {
+            String errorMsg = feature.getName() + ": feature query is potentially wrong as no hits were found";
+            suggestedChangesMap.computeIfAbsent("feature_attributes", k -> new ArrayList<>()).add(errorMsg);
+        }
+        if (counter.get() == anomalyDetector.getFeatureAttributes().size()) {
+            if (!suggestedChangesMap.isEmpty()) {
+                validateAnomalyDetectorResponse();
+            } else {
+                intervalValidation();
+            }
+        }
+    }
+
+    private long[][] createNewTimeRange(long detectorInterval) {
```

Comment:
Can you add comment explaining what does the output mean? Also, you may use some `DateRange` class to hold `start/end` time instead of one array.

Reply:
Added a comment to better explain this method

Reply:
you may refactor this method using `DateTimeRange`

Reply:
Done